### PR TITLE
Add users.lookupByEmail facet to the Web API client

### PR DESCRIPTION
**Pull request by DominikPalo**
###  Summary

Add a missing `users.lookupByEmail` facet to the Web API client (https://api.slack.com/methods/users.lookupByEmail)

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by aoberoi** [comment]
this looks good to me except for the linting errors and the missing `opts` argument.

File: lib/clients/web/facets/users.js [hunk 1]
```diff
@@ -75,6 +76,21 @@ UsersFacet.prototype.list = function list(opts, optCb) {
...
 };
 
+/**
+ * Find a user with an email address.
+ * @see {@link https://api.slack.com/methods/users.lookupByEmail|users.lookupByEmail}
+ *
+ * @param {?} email - An email address belonging to a user in the workspace
+ * @param {function=} optCb Optional callback, if not using promises.
+ */
+UsersFacet.prototype.lookupByEmail = function lookupByEmail(email, optCb) {
```
Comment (aoberoi) [inline lib/clients/web/facets/users.js]: can you add an `opts` argument, and pass it through to `makeAPICall()`?

we've been facing a number of issues that stem from Web API methods that gain optional arguments after they are first released, and then we have to break the API to update that method to include another argument.

**Comment by DominikPalo** [comment]
Thanks, done :)

**Comment by aoberoi** [comment]
Thanks!

**Pull request merged**